src/mpid/common/shm/mpidu_shm.h

```diff
@@ -10,6 +10,7 @@
 
 #define MPIDU_SHM_MAX_FNAME_LEN 256
 #define MPIDU_SHM_CACHE_LINE_LEN 64
+#define MPIDU_SHM_SEG_PROC_LEN (MPIDU_SHM_CACHE_LINE_LEN * 16)
 
```

Comment:
Create a separate `mpidu_init_shm.h` (and `mpidu_init_shm.c` ) and use `MPIDU_INIT_SHM_` prefix.

Declare a struct -- `MPIDU_INIT_SHM_BLOCK_t` -- then you can just use `sizeof(MPIDU_INIT_SHM_BLOCK_t)` for this length. We can add padding to inflate above cache line, or we can add alignment attribute; but since it is init time code, having false sharing may not be a big deal anyway.

Since we are at it, define the init_shm name here as well, e.g. `#define MPIDU_INIT_SHM_NAME  "/mpich_init"`.

Reply:
@hzhou hopefully I have addressed your comments